src/com/dotmarketing/portlets/htmlpages/model/HTMLPage.java

```diff
@@ -62,7 +62,7 @@
     
     private String templateId;
     
-    private long languageId = APILocator.getLanguageAPI().getDefaultLanguage().getId();
```

Comment:
When hibernate inits' it creates an newInstance() of all the mapped classes.  Having this here forced a recursive hibernate call for language, which then called hibernate's buildSessionFactory() which then tried to newInstance the HTMLPage again which....